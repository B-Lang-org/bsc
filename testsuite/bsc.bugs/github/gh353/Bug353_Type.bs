package Bug353_Type where

data Foo a b = Foo b
