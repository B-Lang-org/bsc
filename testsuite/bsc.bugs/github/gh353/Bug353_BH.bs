package Bug353_BH where

import Bug353_Type


struct Foo =
    f1 :: Bool
    f2 :: Bool
  deriving (Bits)


-- -----
-- Trigger error T0007

fn1 :: Foo
fn1 = let res = Foo { f1 = True; f2 = False; };
      in res


-- -----
-- Trigger error T0080

fn2 :: Foo
fn2 = Foo { f1 = True; f2 = False; }
