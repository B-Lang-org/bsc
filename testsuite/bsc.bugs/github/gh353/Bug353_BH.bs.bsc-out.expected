checking package dependencies
compiling Bug353_BH.bs
Error: "Bug353_BH.bs", line 16, column 16: (T0007)
  Unbound type constructor `Bug353_Type.Foo_$Foo'
Error: "Bug353_BH.bs", line 24, column 6: (T0080)
  Type error at the use of the following function:
    Foo

  The expected return type of the function:
    Bug353_BH.Foo

  The return type according to the use:
    Bug353_Type.Foo b__ a__

